### ISSUE TITLE
Add a decay option to `get_scheduler` and refactor to use SequentialLR

### ISSUE THREAD

**Issue by chanind** [jbloomAus/mats_sae_training#61]
As a follow-up to #57, we should refactor the `get_scheduler()` function to make it possible to add a linear warm-up and decay to any scheduler. The [SequentialLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.SequentialLR.html#sequentiallr) scheduler should be able to help with this, and should make the function simpler as well.
